pkg/pools/pools.go

```diff
@@ -86,7 +90,12 @@ func Copy(dst io.Writer, src io.Reader) (written int64, err error) {
 // NewReadCloserWrapper returns a wrapper which puts the bufio.Reader back
 // into the pool and closes the reader if it's an io.ReadCloser.
 func (bufPool *BufioReaderPool) NewReadCloserWrapper(buf *bufio.Reader, r io.Reader) io.ReadCloser {
+	var closed int32 = 0
 	return ioutils.NewReadCloserWrapper(r, func() error {
+		if !atomic.CompareAndSwapInt32(&closed, 0, 1) {
```

Comment:
ha! I was thinking about atomic bool when I started looking (but that led me down the other question I had) 😂